enclave_manager/avalon_enclave_manager/base_enclave_manager.py

```diff
@@ -0,0 +1,275 @@
+    def start_enclave_manager(self):
+        """
+        Executes boot flow and run time flow. This is the engine of an enclave
+        manager which keeps running as long as there are no unexpected errors.
+        It should first carry out the boot time flow followed by the run time
+        routine which is indefinite.
+        """
+        pass
+
+# -------------------------------------------------------------------------
+
+    def _connect_to_kv_store(self):
+        """
+        This function creates a connection to the KVStorage.
+
+        Returns :
+            kv_helper - An instance of LMDBHelperProxy that helps interact
+                        with the database
+        """
+        if self._config.get("KvStorage") is None:
+            logger.error("Kv Storage path is missing")
+            sys.exit(-1)
+        try:
+            kv_helper = connector.open(self._config['KvStorage']['remote_url'])
+        except Exception as err:
+            logger.error("Failed to open KV storage interface; " +
+                         "exiting Intel SGX Enclave manager: {}".format(err))
+            sys.exit(-1)
+        return kv_helper
+
+# -------------------------------------------------------------------------
+
+    def _setup_enclave(self):
+        """
+        This function is responsile for initialization of the trusted
+        enclave. It does so via the EnclaveInfo class which not only
+        initializes the enclave but also ensure the enclave creates
+        the signup data. This signup data is key to all the trusted
+        functionalities over this instance of the enclave. Therefore
+        all relevant data is sent back here. This is kept in memory
+        for use hereafter.
+
+        Returns :
+            signup_data - Relevant signup data to be used for requests to the
+                          enclave
+            extended_measurements - A tuple of enclave basename & measurements
+        """
+        try:
+            logger.info("Initialize enclave and create signup data")
+            signup_data = enclave_info.\
+                EnclaveInfo(self._config.get("EnclaveModule"))
+            # Extended measurements is a list of enclave basename and
+            # enclave measurement
+            extended_measurements = signup_data.get_extended_measurements()
+        except Exception as e:
+            logger.exception("failed to initialize/signup enclave; %s", str(e))
+            sys.exit(-1)
+        return signup_data, extended_measurements
+
+# -----------------------------------------------------------------
+
+
+def validate_request(wo_request):
+    """
+    Validate JSON workorder request
+    """
+    try:
+        json.loads(wo_request)
+    except ValueError as e:
+        logger.error("Invalid JSON format found for workorder - %s", e)
+        return False
+    return True
+
+# -----------------------------------------------------------------
+
+
+def create_json_worker(enclave_data, config):
```

Comment:
If only derived class make use of these methods then better to include in class.

Reply:
One of the derived classes will use and another will not. Moreover it was also not involved in any logical flow that is to say it only prepares a json extracting data from enclave_data. This can be doable even outside an instance of the Enclave Manager. Hence, I let it belong to the module only.